bindings/python/src/lib.rs

```diff
@@ -544,6 +501,84 @@ impl Open {
         }
     }
 
+    /// Return the special non tensor information in the header
+    ///
+    /// Returns:
+    ///     (`Dict[str, str]`):
+    ///         The freeform metadata.
+    pub fn metadata(&self) -> Option<HashMap<String, String>> {
+        self.metadata.metadata().clone()
+    }
+
+    /// Returns the names of the tensors in the file.
+    ///
+    /// Returns:
+    ///     (`List[str]`):
+    ///         The name of the tensors contained in that file
+    pub fn keys(&self) -> PyResult<Vec<String>> {
+        let mut keys: Vec<String> = self.metadata.tensors().keys().cloned().collect();
+        keys.sort();
+        Ok(keys)
+    }
+
+    /// Returns a full tensor
+    ///
+    /// Args:
+    ///     name (`str`):
+    ///         The name of the tensor you want
+    ///
+    /// Returns:
+    ///     (`Tensor`):
+    ///         The tensor in the framework you opened the file for.
+    ///
+    /// Example:
+    /// ```python
+    /// from safetensors import safe_open
+    ///
+    /// with safe_open("model.safetensors", framework="pt", device=0) as f:
+    ///     tensor = f.get_tensor("embedding")
+    ///
+    /// ```
+    pub fn get_tensor(&self, name: &str) -> PyResult<PyObject> {
+        let tensors = self.metadata.tensors();
+        let info = tensors.get(name).ok_or_else(|| {
+            SafetensorError::new_err(format!("File does not contain tensor {name}",))
+        })?;
+        self.hydrate_tensor(info)
+    }
+
+    /// Returns a list of selected tensors
```

Comment:
Was going to implement it in a way that returns all tensors, but filtering also works perfectly fine. For cases where loading N tensors puts a lot of pressure to RAM (with the end goal of them shifting them to VRAM), one can do batching with `get_tensors()` that loads a batch of tensors, moves them to GPU, and then repeats.